R/render.R

```diff
@@ -197,6 +197,33 @@ render_quarto_lang <- function(language_code, path, output_dir, type) {
   config_path <- file.path(temporary_directory, project_name, "_quarto.yml")
   config <- read_yaml(config_path)
 
+  freeze_folder_exists <- fs::dir_exists(
+    file.path(temporary_directory, project_name, "_freeze")
+  )
+
+  if (freeze_folder_exists) {
+    freeze_path <- fs::path(temporary_directory, project_name, "_freeze")
+    freeze_temp <- fs::path(
+      temporary_directory,
+      project_name, paste0("_freeze.", language_code)
+    )
+    freeze_ls <- fs::dir_ls(freeze_path, recurse = TRUE)
+
+    freeze_lang <- freeze_ls[grep(paste0("\\.", language_code, "$"), freeze_ls)]
+    freeze_dirs <- fs::path_rel(freeze_lang, start = freeze_path)
+    freeze_dirs <- gsub(
+      paste0(".", language_code), "",
+      freeze_dirs, fixed = TRUE
+    )
+
+    fs::dir_copy(freeze_lang, fs::path(freeze_temp, freeze_dirs))
```

Comment:
should all of this go into a well-named helper function?

Reply:
I'm not sure I follow. You're suggesting a helper function for the copy and move/delete part, or the entire logic around freeze folders (L205-L224)?

Reply:
for the entire logic around freeze folders **if you agree it's a good idea**, only a suggestion :smile_cat:

Reply:
Yes, I agree! It will help us avoid lintr's cyclomatic complexity linter 😅